playbooks/robusta_playbooks/job_restart_on_oomkilled.py

```diff
@@ -0,0 +1,247 @@
+
+            if req_memory < max_resource:
+
+                keep_the_same = False
+                containers.append(
+                    increase_resource(
+                        container,
+                        max_resource,
+                        params.increase_by,
+                        keep_the_same,
+                    )
+                )
+            else:
+                print("call")
+                finding.title = f"MAX REACHED"
+                finding.add_enrichment(
+                    [
+                        MarkdownBlock(
+                            f"*container request memory has reached the limit*\n```\n{container.name}\n```"
+                        ),
+                    ]
+                )
+
+                keep_the_same = True
+                containers.append(
+                    increase_resource(
+                        container,
+                        max_resource,
+                        params.increase_by,
+                        keep_the_same,
+                    )
+                )
+        elif container.name in running_containers:
+            keep_the_same = True
+            containers.append(
+                increase_resource(
+                    container,
+                    max_resource,
+                    params.increase_by,
+                    keep_the_same,
+                )
+            )
+
+    job_spec = job_fields(job_event, containers)
+    job_event.delete()
+    job_spec.create()
+
+    containers_memory_list = []
+
+    # Getting information for finding
+    for index, containers in enumerate(job_spec.spec.template.spec.containers):
+        containers_memory_list.append(containers.name)
+        containers_memory_list.append(containers.resources.requests["memory"])
+
+    finding.add_enrichment(
+        [
+            MarkdownBlock(
+                f"*containers memory after restart*\n```\n{containers_memory_list}\n```"
+            ),
+        ]
+    )
+    event.add_finding(finding)
+
+
+# Function to increase resource of the container
+def increase_resource(container, max_resource, increase_by, keep_the_same):
+    # Getting Container attributes
+    container = Container(
+        name=container.name,
+        image=container.image,
+        livenessProbe=container.livenessProbe,
+        securityContext=container.securityContext,
+        args=container.args,
+        command=container.command,
+        ports=container.ports,
+        lifecycle=container.lifecycle,
+        readinessProbe=container.readinessProbe,
+        workingDir=container.workingDir,
+        env=container.env,
+        startupProbe=container.startupProbe,
+        envFrom=container.envFrom,
+        imagePullPolicy=container.imagePullPolicy,
+        resources=memory_increment(
+            container.resources,
+            increase_by,
+            max_resource,
+            keep_the_same,
+        )
+        if (container.resources.limits and container.resources.requests)
```

Comment:
if the container does not have limits and requests, will this cause an infinite restarts for the job?

Reply:
The purpose of this condition is to check if the container has resource limits and requests defined. The script is trying to increase the requested memory of the container, and it needs to make sure that the container has memory limits and requests defined before trying to increase it. If the container does not have memory limits and requests defined, the script will not be able to increase the requested memory and it will return None.